### PR TITLE
Implements periodic action

### PR DESCRIPTION
**Pull request by Ekscentricitet**
Added periodic action to the automation events which implements proposal number two from https://github.com/BartoszCichecki/LenovoLegionToolkit/issues/944. The action utilizes the TimeAutomationEvent. The period is calculated from the start of the day.

## Summary by CodeRabbit

- **New Features**
	- Introduced periodic automation triggers, enabling users to schedule automation actions at specified time intervals.
	- Added user interface components for configuring and managing periodic triggers within automation pipelines.
- **Enhancements**
	- Improved automation environment with enhanced properties for more precise control over periodic actions.
	- Enhanced trigger condition checks to initiate time-based events based on trigger type.
- **Documentation and Localization**
	- Updated resource files to include new entries for "Periodic action" and associated UI elements, enhancing localization capabilities.

### CODE REVIEW

**Comment by Ekscentricitet** [comment]
General note: I was a lot of trial and error to implement this. For me it was fun to figure it out and I understand the code base better now. However, I was wondering if some documentation won't be useful for new contributors?

File: LenovoLegionToolkit.WPF/Controls/Automation/AutomationPipelineControl.cs [hunk 1]
```diff
@@ -290,6 +290,9 @@ private string GenerateSubtitle()
         if (AutomationPipeline.Trigger is IWiFiConnectedPipelineTrigger wt && wt.Ssids.Any())
             result += $" | {string.Join(",", wt.Ssids)}";
 
+        if (AutomationPipeline.Trigger is IPeriodicAutomationPipelineTrigger pet && pet.PeriodMinutes != null)
+            result += $" | {string.Join(": ", Resource.PeriodicActionPipelineTriggerTabItemContent_PeriodMinutes, pet.PeriodMinutes.ToString())}";
```
Comment (BartoszCichecki) [inline LenovoLegionToolkit.WPF/Controls/Automation/AutomationPipelineControl.cs:294]: String interpolation would be enough here:

`result += $" | {Resource.PeriodicActionPipelineTriggerTabItemContent_PeriodMinutes}: {pet.PeriodMinutes}";`

File: LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/IAutomationPipelineTrigger.cs [hunk 2]
```diff
@@ -79,3 +79,9 @@ public interface IWiFiConnectedPipelineTrigger : IAutomationPipelineTrigger
 }
 
 public interface IWiFiDisconnectedPipelineTrigger : IDisallowDuplicatesAutomationPipelineTrigger { }
+
+public interface IPeriodicAutomationPipelineTrigger : IAutomationPipelineTrigger 
+{
+    public int? PeriodMinutes { get; }
```
Comment (BartoszCichecki) [inline LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/IAutomationPipelineTrigger.cs:85]: Avoid optionals when not needed, I would rather default this to `0` or something.

File: LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/PeriodicAutomationPipelineTrigger.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+using System.Threading.Tasks;
+using LenovoLegionToolkit.Lib.Automation.Resources;
+
+namespace LenovoLegionToolkit.Lib.Automation.Pipeline.Triggers
+{
+    public class PeriodicAutomationPipelineTrigger : IPeriodicAutomationPipelineTrigger
+    {
+        public string DisplayName => Resource.PeriodicActionPipelineTrigger_DisplayName;
+
+        public int? PeriodMinutes { get; }
```
Comment (BartoszCichecki) [inline LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/PeriodicAutomationPipelineTrigger.cs:12]: Same re: optionals

File: LenovoLegionToolkit.WPF/Extensions/AutomationPipelineTriggerExtensions.cs [hunk 4]
```diff
@@ -19,6 +19,7 @@ public static class AutomationPipelineTriggerExtensions
         IOnStartupAutomationPipelineTrigger => SymbolRegular.Flash24,
         IWiFiConnectedPipelineTrigger => SymbolRegular.Wifi124,
         IWiFiDisconnectedPipelineTrigger => SymbolRegular.WifiOff24,
+        IAutomationPipelineTrigger => SymbolRegular.ArrowRepeatAll24,
```
Comment (BartoszCichecki) [inline LenovoLegionToolkit.WPF/Extensions/AutomationPipelineTriggerExtensions.cs:22]: This shoudl rather be `IPeriodicAutomationPipelineTrigger`

File: LenovoLegionToolkit.WPF/Windows/Automation/AutomationPipelineTriggerConfigurationWindow.xaml.cs [hunk 5]
```diff
@@ -101,6 +101,7 @@ private void CancelButton_Click(object sender, RoutedEventArgs e)
         IUserInactivityPipelineTrigger ut when ut.InactivityTimeSpan > TimeSpan.Zero => true,
         ITimeAutomationPipelineTrigger => true,
         IWiFiConnectedPipelineTrigger => true,
+        IAutomationPipelineTrigger => true,
```
Comment (BartoszCichecki) [inline LenovoLegionToolkit.WPF/Windows/Automation/AutomationPipelineTriggerConfigurationWindow.xaml.cs:104]: Same here: `IPeriodicAutomationPipelineTrigger` and similar to line 101, you can add a condition:

`IPeriodicAutomationPipelineTrigger papt when papt.PeriodMinutes > 0`

File: LenovoLegionToolkit.WPF/Windows/Automation/CreateAutomationPipelineWindow.xaml.cs [hunk 6]
```diff
@@ -40,7 +40,8 @@ public partial class CreateAutomationPipelineWindow
         new WiFiConnectedAutomationPipelineTrigger(Array.Empty<string>()),
         new WiFiDisconnectedAutomationPipelineTrigger(),
         new TimeAutomationPipelineTrigger(false, false, TimeExtensions.UtcNow, Enum.GetValues<DayOfWeek>()),
-        new OnStartupAutomationPipelineTrigger()
+        new OnStartupAutomationPipelineTrigger(),
+        new PeriodicAutomationPipelineTrigger(null)
```
Comment (BartoszCichecki) [inline LenovoLegionToolkit.WPF/Windows/Automation/CreateAutomationPipelineWindow.xaml.cs:44]: This array also defines order on the UI (I know, not the most obvious and best design), but I would place it one row up, so it appears next to "At specified time" trigger on the list.

File: LenovoLegionToolkit.WPF/Windows/Automation/TabItemContent/PeriodicActionPipelineTriggerTabItemContent.xaml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+            <ui:NumberBox
+          x:Name="_periodPickerMinutes"
+          ClearButtonEnabled="False"
+          MaxDecimalPlaces="0"
+          Maximum="1440"
+          Minimum="1"
+          Grid.Column="1" />
+    </Grid>
+   </StackPanel>
+</UserControl>
```
Comment (BartoszCichecki) [inline LenovoLegionToolkit.WPF/Windows/Automation/TabItemContent/PeriodicActionPipelineTriggerTabItemContent.xaml:43]: Missing empty line at the end (makes diffs less readable).

File: LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/IAutomationPipelineTrigger.cs (hunk 2 above)
Comment (BartoszCichecki) [inline LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/IAutomationPipelineTrigger.cs:85]: I would consider using `TimeSpan` to store the value - will be more future proof.

**Review by BartoszCichecki — CHANGES_REQUESTED** [review]

File: LenovoLegionToolkit.WPF/Controls/Automation/AutomationPipelineControl.cs [hunk 8]
```diff
@@ -290,6 +290,9 @@ private string GenerateSubtitle()
         if (AutomationPipeline.Trigger is IWiFiConnectedPipelineTrigger wt && wt.Ssids.Any())
             result += $" | {string.Join(",", wt.Ssids)}";
 
+        if (AutomationPipeline.Trigger is IPeriodicAutomationPipelineTrigger pet && pet.PeriodMinutes != null)
+            result += $" | {Resource.PeriodicActionPipelineTriggerTabItemContent_PeriodMinutes}: {pet.PeriodMinutes}";
```
[the remark this answers is not in the document]
Comment (BartoszCichecki) [reply LenovoLegionToolkit.WPF/Controls/Automation/AutomationPipelineControl.cs:294]: `is` operator will return false if `AutomationPipeline.Trigger` is null, therefore checking if `pet` is null isn't necessary.

**Review by BartoszCichecki — APPROVED** [review]

**Pull request merged**